libdnf/module/module_db.hpp

```diff
@@ -0,0 +1,109 @@
+    /// Get installed profiles for given module name.
+    const std::vector<std::string> & get_installed_profiles(const std::string & module_name) const;
+
+    /// Get module names of all modules that were not originally disabled, but are disabled now.
+    std::vector<std::string> get_all_newly_disabled_modules() const;
+    /// Get module names of all modules that were not originally available, but are available now.
+    std::vector<std::string> get_all_newly_reset_modules() const;
+    /// Get map of module names and their enabled streams for all modules that were not originally enabled,
+    /// but are enabled now.
+    std::map<std::string, std::string> get_all_newly_enabled_streams() const;
+    /// Get map of module names and their originally enabled streams for all modules that were originally enabled,
+    /// but are disabled now.
+    std::map<std::string, std::string> get_all_newly_disabled_streams() const;
+    /// Get map of module names and their originally enabled streams for all modules that were originally enabled,
+    /// but are available now.
+    std::map<std::string, std::string> get_all_newly_reset_streams() const;
+    /// Get map of module names and their originally and newly enabled streams for all modules with switched streams.
+    std::map<std::string, std::pair<std::string, std::string>> get_all_newly_switched_streams() const;
+    /// Get map of module names and their newly installed profiles for all modules with added profiles.
+    std::map<std::string, std::vector<std::string>> get_all_newly_installed_profiles() const;
+    /// Get map of module names and their removed profiles for all modules with removed profiles.
+    std::map<std::string, std::vector<std::string>> get_all_newly_removed_profiles() const;
+
+    bool change_status(const std::string & module_name, ModuleStatus status);
+    bool change_stream(const std::string & module_name, const std::string & stream, const bool count = false);
+    bool add_profile(const std::string & module_name, const std::string & profile);
+    bool remove_profile(const std::string & module_name, const std::string & profile);
+    void clear_profiles(const std::string & module_name);
+
+private:
+    friend ModuleSack;
+
+    struct RuntimeModuleState {
+        system::ModuleState original;
+        system::ModuleState changed;
+        int stream_changes_num = 0;
+    };
+
+
+    /// Get the runtime module state for given module name.
+    const RuntimeModuleState & get_runtime_module_state(const std::string & module_name) const;
+    RuntimeModuleState & get_runtime_module_state(const std::string & module_name);
+
+    /// Get all module names for modules that newly have the given status.
+    std::vector<std::string> get_all_changed_modules(const ModuleStatus new_status) const;
+    /// Get all module names and changes in their profiles if there are any.
+    std::map<std::string, std::vector<std::string>> get_all_changed_profiles(bool installed = true) const;
```

Comment:
I am sorry but installed argument is not described.

Reply:
Fixed.